### PR TITLE
Pamd Module

### PR DESCRIPTION
**Pull request by kevensen**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

pamd
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### SUMMARY

While lineinfile and replace are very useful, I felt that a module specific to updating pam.d services was necessary.

### CODE REVIEW

**Comment by kevensen** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/2640

**Comment by kevensen** [comment]
I know this request is old, but I really would like it incorporated.

File: lib/ansible/modules/system/pamd.py [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+DOCUMENTATION = """
+module: pamd
+author:
+    - "Kenneth D. Evensen (@kevensen)"
+short_description: Manage PAM Modules
+description:
+  - Edit PAM service's type, control, module path and module arguments.
+    In order for a PAM rule to be modified, the type, control and
+    module_path must match an existing rule.  See man(5) pam.d for details.
+version_added: "2.2"
```
Comment (gundalow) [inline lib/ansible/modules/system/pamd.py]: 2.3

File: lib/ansible/modules/system/pamd.py [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+    default: /etc/pam.d/
+    description:
+    - This is the path to the PAM service files
+"""
+
+EXAMPLES = """
+- name: Update pamd rule's control in /etc/pam.d/system-auth
+  pamd:
+    name=system-auth
+    type=auth
```
Comment (gundalow) [inline lib/ansible/modules/system/pamd.py]: For examples we are standardizing on
`key: value`
Rather than key=value

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by kevensen** [comment]
@gundalow I've made the changes you requested.  At your convenience, could you please take another look?

**Comment by gundalow** [comment]
Thank you for being very proactive in your efforts to migrate your PR, however it appears that there was a window of time when ansible/ansible:devel was broken and it appears, unfortunately, that your PR was migrated during that window.

Due to this you will need to follow a hopefully very simple process to fix this. Please see https://groups.google.com/forum/#!topic/ansible-devel/0eRSpiGkios

As a thank you for being pioneers, we will actively review your PRs as a priority. Once you've migrated your PR please add it to the Core Agenda https://github.com/ansible/community/issues/147

**Pull request closed without merging**